### PR TITLE
Adding basic side detection and propagation.

### PR DESCRIPTION
**Pull request by ptpham**
This might not be a complete set of mechanisms to compute the sides information, but it's a start. XD

### CODE REVIEW

**Comment by ptpham** [comment]
Hold on ... totally gonna patch this or something. XD

**Comment by ptpham** [comment]
Okay, I think this is something more sensible. XD

File: src/relation.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 
+var _ = require('lodash');
 var glMatrix = require('./gl-matrix');
 var v3 = glMatrix.vec3;
 var quat = glMatrix.quat;
 var geometry = require('./geometry');
+var Multimap = require('multimap');
```
Comment (ptpham) [inline src/relation.js]: Goood point.

**Comment by cixzhang** [comment]
Looks goooood

**Pull request merged**